std::to_string is not a member of 'std'

## Description 

It's my understanding that this is due to [cygwin issues](https://stackoverflow.com/questions/33500074/cygwin-g-5-2-to-string-is-not-a-member-of-std)
Mailing list: [here](https://cygwin.com/ml/cygwin/2015-10/msg00421.html)

I cannot modify my gcc:cygwin. 

### Steps to reproduce
<!--
This is only relevant for bug reports, but if you do have one,
please provide a minimal set of steps to reproduce the problem.

Usually this means providing a small and self-contained code using Catch
and specifying compiler flags/tools used if relevant.
-->

```powershell
PS C:\application\common\tests> g++ -std=c++11 -Wfatal-errors -Wall -I. -c 000-CatchMain.cpp
In file included from 000-CatchMain.cpp:9:0:
catch.hpp: In function 'std::string Catch::to_string(const T&)':
catch.hpp:8418:16: error: 'to_string' is not a member of 'std'
         return std::to_string(t);
                ^
compilation terminated due to -Wfatal-errors.
```


### Extra information
<!--
Fill in any extra information that might be important for your issue.

If your issue is a bugreport, definitely fill out at least the following.
-->
```powershell
PS C:\application\common\tests> head -n 3 .\catch.hpp
/*
 *  Catch v2.4.0
 *  Generated: 2018-09-04 11:55:01.682061
```

```cmd
C:\Users\me>ver

Microsoft Windows [Version 10.0.16299.611]
```


```powershell
PS C:\application\common\tests> g++ --version
g++ (GCC) 4.9.3
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/include/internal/catch_compiler_capabilities.h
+++ b/include/internal/catch_compiler_capabilities.h
@@ -109,7 +109,14 @@
 // Required for some versions of Cygwin to declare gettimeofday
 // see: http://stackoverflow.com/questions/36901803/gettimeofday-not-declared-in-this-scope-cygwin
 #   define _BSD_SOURCE
+// some versions of cygwin (most) do not support std::to_string. Use the libstd check. 
+// https://gcc.gnu.org/onlinedocs/gcc-4.8.2/libstdc++/api/a01053_source.html line 2812-2813
+# if !((__cplusplus >= 201103L) && defined(_GLIBCXX_USE_C99) \
+	       && !defined(_GLIBCXX_HAVE_BROKEN_VSWPRINTF))
 
+#	define CATCH_INTERNAL_CONFIG_NO_CPP11_TO_STRING
+
+# endif
 #endif // __CYGWIN__
 
 ////////////////////////////////////////////////////////////////////////////////

